tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -1178,6 +1213,139 @@ bool CorProfiler::ProfilerAssemblyIsLoadedIntoAppDomain(AppDomainID app_domain_i
+        Logger::Info("sizeof(pulHashAlgId): ", sizeof(managed_profiler_assembly_property.pulHashAlgId));
+        Logger::Info("assemblyFlags: ", managed_profiler_assembly_property.assemblyFlags);
+    }
+
+    //
+    // *** Get DistributedTracer TypeDef
+    //
+    mdTypeDef distributedTracerTypeDef;
+    hr = module_metadata.metadata_import->FindTypeDefByName(WStr("Datadog.Trace.ClrProfiler.DistributedTracer"),
+                                                            mdTokenNil, &distributedTracerTypeDef);
+    if (FAILED(hr))
+    {
+        Logger::Warn("Error Rewriting for Distributed Tracing on getting DistributedTracer TypeDef");
+        return hr;
+    }
+
+    //
+    // *** Import Current Version of assembly
+    //
+    mdAssemblyRef managed_profiler_assemblyRef;
+    hr = module_metadata.assembly_emit->DefineAssemblyRef(
+        managed_profiler_assembly_property.ppbPublicKey, managed_profiler_assembly_property.pcbPublicKey,
+        managed_profiler_assembly_property.szName.data(), &managed_profiler_assembly_property.pMetaData,
+        &managed_profiler_assembly_property.pulHashAlgId, sizeof(managed_profiler_assembly_property.pulHashAlgId),
+        managed_profiler_assembly_property.assemblyFlags, &managed_profiler_assemblyRef);
+
+    if (FAILED(hr) || managed_profiler_assemblyRef == mdAssemblyRefNil)
+    {
+        Logger::Warn("Error Rewriting for Distributed Tracing on getting ManagedProfiler AssemblyRef");
+        return hr;
+    }
+
+    mdTypeRef distributedTracerTypeRef;
+    hr = module_metadata.metadata_emit->DefineTypeRefByName(
+        managed_profiler_assemblyRef, WStr("Datadog.Trace.ClrProfiler.DistributedTracer"), &distributedTracerTypeRef);
+
+    if (FAILED(hr))
+    {
+        Logger::Warn("Error Rewriting for Distributed Tracing on getting DistributedTracer TypeRef");
+        return hr;
+    }
+
+    //
+    // *** GetDistributedTrace MethodDef ***
+    //
+    COR_SIGNATURE getDistributedTracerSignature[] = {IMAGE_CEE_CS_CALLCONV_DEFAULT, 0, ELEMENT_TYPE_OBJECT};
+    mdMethodDef getDistributedTraceMethodDef;
+    hr = module_metadata.metadata_import->FindMethod(distributedTracerTypeDef, WStr("GetDistributedTracer"),
+                                                     getDistributedTracerSignature, 3, &getDistributedTraceMethodDef);
+    if (FAILED(hr))
+    {
+        Logger::Warn("Error Rewriting for Distributed Tracing on getting GetDistributedTracer MethodDef");
+        return hr;
+    }
+
+    //
+    // *** GetDistributedTrace MemberRef ***
+    //
+    mdMemberRef getDistributedTraceMemberRef;
+    hr =
+        module_metadata.metadata_emit->DefineMemberRef(distributedTracerTypeRef, WStr("GetDistributedTracer"),
+                                                       getDistributedTracerSignature, 3, &getDistributedTraceMemberRef);
+    if (FAILED(hr))
+    {
+        Logger::Warn("Error Rewriting for Distributed Tracing on getting GetDistributedTracer MemberRef");
```

Comment:
```suggestion
        Logger::Warn("Error Rewriting for Distributed Tracing on defining GetDistributedTracer MemberRef");
```

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/2020/commits/e1b26a25a9095a2a389a61a409e792c1b84a20b4